src/H5.c

```diff
@@ -146,6 +146,8 @@ H5_init_library(void)
     size_t i;
     herr_t ret_value = SUCCEED;
 
+    FUNC_ENTER_NOAPI(FAIL)
```

Comment:
Since FUNC_ENTER... can do things like pushing function names where FUNC_LEAVE... expects symmetrical behavior, FUNC_ENTER... always needs to be called before any code that might cause a goto.